kci-dev/subcommands/checkout.py

```diff
@@ -50,6 +54,97 @@ def send_checkout_full(baseurl, token, **kwargs):
     return response.json()
 
 
+def retrieve_treeid_nodes(baseurl, token, treeid):
+    url = baseurl + "latest/nodes/fast?treeid=" + treeid
+    headers = {
+        "Content-Type": "application/json; charset=utf-8",
+        "Authorization": f"{token}",
+    }
+    try:
+        response = requests.get(url, headers=headers)
+    except requests.exceptions.RequestException as e:
+        click.secho(f"API connection error: {e}, retrying...", fg="yellow")
+        return None
+    except Exception as e:
+        click.secho(f"API connection error: {e}, retrying...", fg="yellow")
+        return None
+
+    if response.status_code >= 400:
+        display_api_error(response)
+        return None
+
+    return response.json()
+
+
+def check_node(node):
+    """
+    Node can be defined RUNNING/DONE/FAIL based on the state
+    Simplify, as our current state model suboptimal
+    """
+    name = node["name"]
+    state = node["state"]
+    result = node["result"]
+    if name == "checkout":
+        if state == "running":
+            return "RUNNING"
+        elif state == "available" or state == "closing":
+            return "DONE"
+        elif state == "done" and result == "pass":
+            return "DONE"
+        else:
+            return "FAIL"
+    else:
+        if state == "running":
+            return "RUNNING"
+        elif state == "done" and result == "pass":
+            return "DONE"
+        else:
+            return "FAIL"
+
+
+def watch_jobs(baseurl, token, treeid, jobfilter):
+    # we need to add to jobfilter "checkout" node
+    jobfilter = list(jobfilter)
+    jobfilter.append("checkout")
+    while True:
+        inprogress = 0
+        joblist = jobfilter.copy()
+        nodes = retrieve_treeid_nodes(baseurl, token, treeid)
+        if not nodes:
+            click.secho("No nodes found. Retrying...", fg="yellow")
+            continue
+        time_local = time.localtime()
+        click.echo(f"Current time: {time.strftime('%Y-%m-%d %H:%M:%S', time_local)}")
+        click.secho(f"Total tree nodes {len(nodes)} found.", fg="green")
+
+        # Tricky part in watch is that we might have one item in jobfilter (job, test),
+        # but it might spawn multiple nodes with same name
```

Comment:
why we might spawn multiple nodes with same name?

Reply:
For example baseline-nfs-arm64-qualcomm is usually spawned on all available qualcomm platforms (as they share similar kernel config), so it will be two test nodes, one for trogdor-lazor and one for trogdor-kingoftown.
Each one will have individual node. In some cases it is ok (regression might be reproduced on all devices), in other case i am introducing platformfilter parameter soon, where you can limit device type.